opcua/server/history_sql.py

```diff
@@ -0,0 +1,165 @@
+                                                                                               datavalue.SourceTimestamp,
+                                                                                               datavalue.StatusCode.value,
+                                                                                               str(datavalue.Value.Value),
+                                                                                               datavalue.Value.VariantType.value,
+                                                                                               value_blob))
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Insert Error:', e)
+
+        # get this node's period from the period dict and calculate the limit
+        period = self._datachanges_period[node_id]
+        date_limit = datetime.now() - period
+
+        # after the insert, delete all values older than period
+        try:
+            _c_sub.execute('DELETE FROM "{tn}" WHERE ServerTimestamp < ?'.format(tn=node_id),
+                                                                                (date_limit.isoformat(' '),))
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Delete Old Data Error:', e)
+
+        self._conn.commit()
+
+    def read_node_history(self, node, start, end, nb_values):
+        _c_read = self._conn.cursor()
+
+        if end is None:
+            end = datetime.now() + timedelta(days=1)
+        if start is None:
+            start = ua.DateTimeMinValue
+
+        node_id = self._get_table_name(node)
+
+        cont = None
+        results = []
+
+        start_time = start.isoformat(' ')
+        end_time = end.isoformat(' ')
+
+        # select values from the database
+        try:
+            for row in _c_read.execute('SELECT * FROM "{tn}" WHERE "ServerTimestamp" BETWEEN ? AND ? '
+                                       'LIMIT ?'.format(tn=node_id), (start_time, end_time, nb_values,)):
+
+                value = self._unpack_value(row[4], row[5])
+
+                dv = ua.DataValue(ua.Variant(value, ua.VariantType(row[4])))
+                dv.ServerTimestamp = row[0]
+                dv.SourceTimestamp = row[1]
+                dv.StatusCode = ua.StatusCode(row[2])
+
+                results.append(dv)
+
+        except sqlite3.Error as e:
+            print(node_id, 'Historizing SQL Read Error:', e)
+
+        return results, cont
+
+    def new_historized_event(self, event, period):
+        raise NotImplementedError
+
+    def save_event(self, event):
+        raise NotImplementedError
+
+    def read_event_history(self, start, end, evfilter):
+        raise NotImplementedError
+
+    def _pack_value(self, datavalue):
+        variant_code = datavalue.Value.VariantType.value
+        return struct.pack(self._get_pack_type(variant_code), datavalue.Value.Value)
```

Comment:
OK, this is not necessary. If you want to save variant as binary ust called Variant.to_binary() and variant.from_binary(). No need to use pack directly. 

Would be nice to save data as a python type, but we will get issue with extensions objects, etc,,,,